ReactWindows/ReactNative.Shared/ReactRootView.cs

```diff
@@ -142,20 +186,53 @@ public void StartReactApplication(ReactInstanceManager reactInstanceManager, str
         /// <returns>The desired size.</returns>
         protected override Size MeasureOverride(Size availableSize)
         {
-            DispatcherHelpers.AssertOnDispatcher();
+            DispatcherHelpers.AssertOnDispatcher(this);
 
             var result = base.MeasureOverride(availableSize);
 
+            // Fire and forget async
+            Forget(MeasureOverrideHelperAsync());
+
+            return result;
+        }
+
+        internal void CleanupSafe()
+        {
+            // Inlining allowed
+            DispatcherHelpers.RunOnDispatcher(this.Dispatcher, Cleanup, true);
+        }
+
+        internal void Cleanup()
+        {
+            DispatcherHelpers.AssertOnDispatcher(this);
+
+            Children.Clear();
+            this.ClearData();
+        }
+
+        private async Task MeasureOverrideHelperAsync()
+        {
+            DispatcherHelpers.AssertOnDispatcher(this);
+
             _wasMeasured = true;
 
             var reactInstanceManager = _reactInstanceManager;
             if (_attachScheduled && reactInstanceManager != null)
             {
                 _attachScheduled = false;
-                reactInstanceManager.AttachMeasuredRootView(this);
+
+                await reactInstanceManager.AttachMeasuredRootViewAsync(this);
             }
+        }
 
-            return result;
+        private static void Forget(Task task)
```

Comment:
>Forget [](start = 28, length = 6)

There's probably other places where we're doing `async void` that we may want to use this. Do you think we should break this out for use elsewhere (or just wait until we have a PR with other uses is fine also...)

Reply:
I did have this initially as a TaskExtensions thing, but @rigdern had some reservations with regards to encouraging this use broadly.
At the end I ended up with the method here and one more similar construct in DispatcherHelpers. I personally prefer "task extensions", but I guess Adam should expose his opinion as well.


---
In reply to: [165945530](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r165945530) [](ancestors = 165945530)

Reply:
Let's wait for another use case.

---
In reply to: [166104349](https://github.com/Microsoft/react-native-windows/pull/1615#discussion_r166104349) [](ancestors = 166104349,165945530)